queries.sql

```diff
@@ -6,4 +6,66 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+BEGIN TRANSACTION;
+
+UPDATE animals 
+SET species = 'unspecified';
+
+SELECT * FROM animals
+WHERE species = 'unspecified';
+
+Rollback;
+
+SELECT * FROM animals;
+
+begin;
+
+update animals
+set species = 'digimon'
+where name like '%mon';
+
+update animals 
+set species = 'pokemon'
+where species is null;
+
+select * from animals 
+
+commit;
+
+select * from animals;
+
+begin;
+
+delete from animals;
+
+select * from animals;
+
+rollback;
+
+select * from animals;
+
+begin;
+
+delete from animals
+where date_of_birth > '2022-01-01';
+
+savepoint sp1;
+
+update animals 
+set weight_kg = weight_kg * -1;
+
+commit;
+select * from animals
+
+
+select count(id) from animals;
+select count(escape_attempts) from animals where escape_attempts = 0;
+select neutered, max(escape_attempts) from animals group by neutered;
+select species, max(weight_kg), min(weight_kg) from animals group by species;
+
+select species, avg(escape_attempts) 
+from animals
+where date_of_birth between '1990-01-01' AND '2000-12-31'
+group by species;
```

Comment:
- [ ] Ouch! this is a close one but it seems you forgot to add the query for the question "What is the average weight of animals?", so please make sure to add this later on.